jetstream/maxtext/http-server/http_server.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""HTTP Server to interact with MaxText + JetStream Server."""
+
+import asyncio
+import concurrent.futures
+import json
+import logging
+from typing import Optional
+
+import fastapi
+import grpc
+from jetstream.core.proto import jetstream_pb2
+from jetstream.core.proto import jetstream_pb2_grpc
+import pydantic
+
+
+class GenerateRequest(pydantic.BaseModel):
+  server: Optional[str] = "127.0.0.1"
+  port: Optional[str] = "9000"
+  session_cache: Optional[str] = ""
+  prompt: Optional[str] = "AB"
+  priority: Optional[int] = 0
+  max_tokens: Optional[int] = 3
```

Comment:
Should we add more reasonable defaults for sample prompt and max tokens? We should probably default max_tokens to 100 to see reasonable responses.

Reply:
what is the max_token means here? if we set it to 3, why we still see reasonable response with more than 100 tokens output?

Reply:
users could override it, that's point. If we do need to update all these, then prompt here should be "", per_device_batch_size should be 12

Reply:
Taking another look at this, the sample prompts are in the UG here - https://github.com/GoogleCloudPlatform/ai-on-gke/pull/358/files/75e621e3feb329d34c6502c97d81db97f6ba6f05#diff-e1f2497210c64d2163b6cf01a41229ecd8ed3c4f9304d0d46a6aadab69189e7eR127. These are just defaults if nothing is passed. Assuming that prompt and max tokens are not passed, wouldn't it make more sense to have the response to be expected to be not so good?

Reply:
For the prompt, something like "This is an example prompt" would make more sense because we want to make that clear. Max tokens to 100 so that when someone doesn't pass the max tokens, we will still provide a response of reasonable length instead of terminating at 3 words or less.

> users could override it, that's point

The point of the defaults is to set something that when not provided works reasonably well. Of course users can and will override it. But in the case where someone doesn't, we don't want the defaults to be totally not useful.

Reply:
ok, let's update it to 100 and configure the per_device_batch_size as 12, thanks @vivianrwu

Reply:
Thanks Ashok, changing to your suggestion!